webserver/src/main/java/io/kestra/webserver/controllers/ExecutionController.java

```diff
@@ -753,6 +774,34 @@ public HttpResponse<?> kill(
         return HttpResponse.noContent();
     }
 
+    @ExecuteOn(TaskExecutors.IO)
+    @Post(uri = "executions/{executionId}/resume", produces = MediaType.TEXT_JSON)
+    @Operation(tags = {"Executions"}, summary = "Resume a paused execution.")
+    @ApiResponse(responseCode = "204", description = "On success")
+    @ApiResponse(responseCode = "409", description = "if the executions is not paused")
+    public HttpResponse<?> resume(
+        @Parameter(description = "The execution id") @PathVariable String executionId
+    ) throws InternalException {
+        Optional<Execution> maybeExecution = executionRepository.findById(executionId);
+        if (maybeExecution.isEmpty()) {
+            return HttpResponse.notFound();
+        }
+
+        var execution = maybeExecution.get();
+        if (!execution.getState().isPaused()) {
+            throw new IllegalStateException("Execution is not paused, can't resume it");
+        }
+
+        var runningTaskRun = execution.findFirstByState(State.Type.PAUSED).map(taskRun ->
```

Comment:
Move this logic on a service to have more unit test, example: no pause tasks that should throw an exception and not failed silently

Reply:
I move it to a service but I cannot move the check of paused to the service as if I do this, for the kill method I will check two times if the execution is paused which on some edge case will generate an exception (if the pause task was automatically resume between the two checks). This would result in false negative.